### ISSUE TITLE
.ml indices that are closed prevent Kibana monitoring from displaying.

### ISSUE THREAD

**Issue by travisestill** [elastic/elasticsearch#91893]
### Elasticsearch Version

8.3.3, 8.4.3

### Java Version

_bundled_

### OS Version

any

### Problem Description

.ml indices that are closed prevent Kibana monitoring from displaying. Using the GET _ml/anomaly_detectors/_stats endpoint:

```
{
  "error": {
    "root_cause": [
      {
        "type": "cluster_block_exception",
        "reason": "index [.ml-anomalies-shared] blocked by: [FORBIDDEN/4/index closed];"
      }
    ],
    "type": "cluster_block_exception",
    "reason": "index [.ml-anomalies-shared] blocked by: [FORBIDDEN/4/index closed];"
  },
  "status": 403
}
```

### Steps to Reproduce

(Reproduced)

1. Started with the closed .ml indices. (In ESS) Run a plan to change Logging and Monitoring to send to itself (self-monitoring), not a dedicated monitoring cluster.
2. Plan change completed but the cluster's Stack Monitoring page would not render any data, it shows the message that you need to enable it in the console.
3. Opened the .ml* indices and the Stack Monitoring data is displayed.

**Comment by droberts195** [comment]
This is back from the `beats` repo, with the requirement that `MachineLearningUsageTransportAction` should not propagate exceptions that make X-Pack usage return an error.  We should just miss out the ML section if it cannot be populated.

**Comment by droberts195** [comment]
I have confirmed that #91917 fixes this.

<img width="1471" alt="Screenshot 2022-11-24 at 17 24 27" src="https://user-images.githubusercontent.com/7405510/203840054-d073a5a8-fae4-4957-847c-309fea49a882.png">

Now if you close the ML results index the stack monitoring page still displays, albeit with an incorrect value for the number of ML jobs. I think that's the best that can be expected in the circumstances. Internal features cannot be expected to operate perfectly if their internal state has been changed in unexpected ways. But at least now meddling with the ML internals doesn't _completely_ break stack monitoring.
